test/e2e/framework.go

```diff
@@ -556,6 +556,40 @@ func (data *TestData) podWaitForIPs(timeout time.Duration, name, namespace strin
 	return ips, nil
 }
 
+// deploymentWaitForIPsOrNames polls the K8s apiServer until the specified Pod in deployment has an IP address
+func (data *TestData) deploymentWaitForIPsOrNames(timeout time.Duration, namespace, deployment string) ([]string, []string, error) {
```

Comment:
I think we can merge deploymentWaitForIPsOrNames and deploymentWaitForNames to deploymentWaitFor like podWaitFor/vmWaitFor?

Reply:
I checked the code, `deploymentWaitForNames` is for polling pecific pods are created, no matter they are running or not.
but here, `deploymentWaitForIPsOrNames`  is specific for running state pods, so pods have IPs and names. 
Two functions are close but used in different use cases. 
So, I think we can keep both functions instead of merging into one func.